ext/bindings/message.cpp

```diff
@@ -12,9 +14,32 @@ Rice::String MessageId::toString() {
   return Rice::String(ss.str());
 }
 
-Message::Message(const std::string& data) {
+Message::Message(const std::string& data, Rice::Object arg = Rice::Object()) {
   pulsar::MessageBuilder mb;
   mb.setContent(data);
+
+  if (arg && arg.rb_type() != T_NONE) {
+    Rice::Hash opts = Rice::Hash(arg);
+    Rice::Hash::iterator it = opts.begin();
+    Rice::Hash::iterator end = opts.end();
+    std::string key;
+    for (; it != end; ++it) {
+      key = it->key.to_s().str();
+      if (key == "properties"){
```

Comment:
I think this is fine for now, but for all the options we would eventually support. I wonder if it may make sense to refactor out this method and turn it into a switch statement.